cloud/services/virtualmachines/virtualmachines.go

```diff
@@ -348,6 +348,10 @@ func (s *Service) generateStorageProfile(ctx context.Context, vmSpec azure.VMSpe
 		}
 	}
 
+	if vmSpec.OSDisk.ManagedDisk.DiskEncryptionSet != nil {
+		storageProfile.OsDisk.ManagedDisk.DiskEncryptionSet = &compute.DiskEncryptionSetParameters{ID: to.StringPtr(vmSpec.OSDisk.ManagedDisk.DiskEncryptionSet.ID)}
```

Comment:
let's also add this to scale sets which use the same ManagedDisk type: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/0ccb197364a5ed8623cdf500ebedbc9099609d19/cloud/services/scalesets/vmss.go#L252